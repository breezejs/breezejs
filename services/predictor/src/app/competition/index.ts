/**
 * Breeze predictor competition
 * @module breeze/predictor/competition
 */
export * from './competition.controller';
export * from './competition.dto';
export * from './competition.entity';
export * from './competition.enum';
export * from './competition.module';
export * from './competition.providers';
export * from './competition.service';
