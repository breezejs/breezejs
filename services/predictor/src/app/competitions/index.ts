/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './competitions.module';
