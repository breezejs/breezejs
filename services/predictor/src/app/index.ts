/**
 * Breeze BB predictor
 * @module breeze-bb/predictor
 */
export * from './app.module';
