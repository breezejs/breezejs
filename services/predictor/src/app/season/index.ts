/**
 * Breeze predictor season
 * @module breeze/predictor/season
 */
export * from './season.controller';
export * from './season.dto';
export * from './season.entity';
export * from './season.enum';
export * from './season.module';
export * from './season.providers';
export * from './season.service';
