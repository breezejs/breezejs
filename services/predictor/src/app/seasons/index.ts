/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './seasons.module';
