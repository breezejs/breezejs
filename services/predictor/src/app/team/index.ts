/**
 * Breeze predictor team
 * @module breeze/predictor/team
 */
export * from './team.controller';
export * from './team.dto';
export * from './team.entity';
export * from './team.enum';
export * from './team.module';
export * from './team.providers';
export * from './team.service';
