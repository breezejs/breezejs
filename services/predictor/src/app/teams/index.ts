/**
 * Breeze predictor service teams module
 * @module breeze/predictor/teams
 */
export * from './teams.module';
