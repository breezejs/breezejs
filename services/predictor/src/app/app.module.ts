/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {Module} from '@nestjs/common';
import {SeasonModule} from './season';
import {TeamModule} from './team';

/**
 * Predictor service app module
 */
@Module({
  imports: [
    SeasonModule,
    TeamModule
  ]
})
export class AppModule {}
