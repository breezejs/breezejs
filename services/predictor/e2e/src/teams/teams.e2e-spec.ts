/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {INestApplication, ValidationPipe} from '@nestjs/common';
import {Test, TestingModule} from '@nestjs/testing';
import * as supertest from 'supertest';
import {TeamsModule} from '../../../src/app/teams';

describe('Teams', (): void => {
  let app: INestApplication;
  let id: string;
  let request: supertest.SuperTest<supertest.Request>;

  beforeAll(async (): Promise<void> => {
    const testModule: TestingModule = await Test
      .createTestingModule({
        imports: [
          TeamsModule
        ]
      })
      .compile();

    app = testModule.createNestApplication();

    app.useGlobalPipes(new ValidationPipe());

    await app.init();

    request = supertest(app.getHttpServer());
  });

  afterAll(async (): Promise<void> => {
    await app.close();
  });

  describe('[GET] /teams', (): void => {
    describe('with no filtering or sorting', (): void => {
      it('should list first page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams');

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          page: 3
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });
    });

    describe('with filtering', (): void => {
      it('should list first page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          fields: 'id,name'
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          fields: 'id,name',
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          fields: 'id,name',
          page: 3
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });

      it('should return a bad request if filtered field does not exist', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          fields: 'club'
        });

        expect(status).toEqual(400);
        expect(body).toMatchSnapshot();
      });
    });

    describe('with sorting', (): void => {
      describe('ascending', (): void => {
        it('should list first page of teams with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            sort: 'name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should list second page of teams with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            page: 2,
            sort: 'name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should return no content if out of bounds', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            page: 3,
            sort: 'name'
          });

          expect(status).toEqual(204);
          expect(body).toMatchSnapshot();
        });
      });

      describe('descending', (): void => {
        it('should list first page of teams with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            dir: 'desc',
            sort: 'name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should list second page of teams with all values', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            dir: 'desc',
            page: 2,
            sort: 'name'
          });

          expect(status).toEqual(200);
          expect(body).toMatchSnapshot();
        });

        it('should return no content if out of bounds', async (): Promise<void> => {
          const {body, status}: supertest.Response = await request.get('/teams').query({
            dir: 'desc',
            page: 3,
            sort: 'name'
          });

          expect(status).toEqual(204);
          expect(body).toMatchSnapshot();
        });
      });
    });

    describe('with limiting', (): void => {
      it('should list first page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          limit: 5
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should list second page of teams with all values', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          limit: 5,
          page: 2
        });

        expect(status).toEqual(200);
        expect(body).toMatchSnapshot();
      });

      it('should return no content if out of bounds', async (): Promise<void> => {
        const {body, status}: supertest.Response = await request.get('/teams').query({
          limit: 5,
          page: 4
        });

        expect(status).toEqual(204);
        expect(body).toMatchSnapshot();
      });
    });
  });

  describe('[GET] /teams/:id', (): void => {
    it('should return a team', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/teams/f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d');

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a filtered result', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/teams/f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d').query({
        fields: 'name'
      });

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a bad request if filtered field does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/teams/f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d').query({
        fields: 'club'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if team does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/teams/70340225-c90a-4b9b-a979-2d16212acd16');

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[POST] /teams', (): void => {
    it('should create a new team', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/teams').send({
        name: 'Team'
      });

      id = body.id;

      expect(status).toEqual(201);
      expect(body).toEqual({
        createdAt: expect.any(String),
        id,
        name: 'Team',
        updatedAt: expect.any(String)
      });
    });

    it('should throw an error if a duplicate team is created', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/teams').send({
        name: 'Team'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error if name is not a string', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.post('/teams').send({
        name: 99
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[PATCH] /teams/:id', (): void => {
    it('should update a team', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/teams/${id}`).send({
        name: 'Team rename'
      });

      expect(status).toEqual(204);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error of a team is updated with a duplicate name', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/teams/${id}`).send({
        name: 'AFC Wimbledon'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if team does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch('/teams/88044266-a83a-4e10-b85b-e552790909a1').send({
        name: 'Team rename'
      });

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });

    it('should throw an error if name is not a string', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.patch(`/teams/${id}`).send({
        name: 99
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });
  });

  describe('[DELETE] /teams/:id', (): void => {
    it('should delete a team', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.delete(`/teams/${id}`);

      expect(status).toEqual(204);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if team does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.delete('/teams/88044266-a83a-4e10-b85b-e552790909a1');

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });
  });
});
