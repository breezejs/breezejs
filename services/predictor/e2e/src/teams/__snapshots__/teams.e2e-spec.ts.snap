// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams [DELETE] /teams/:id should delete a team 1`] = `Object {}`;

exports[`Teams [DELETE] /teams/:id should return not found if team does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Teams [GET] /teams with filtering should list first page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "id": "f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d",
      "name": "AFC Wimbledon",
    },
    Object {
      "id": "07c3eac0-909d-4a1c-9fb5-6a6202c65e34",
      "name": "Arsenal",
    },
    Object {
      "id": "ebdcacdb-cdd8-4817-b87b-412c3e83ecc0",
      "name": "Bournemouth",
    },
    Object {
      "id": "47362a27-5245-4492-ab5b-8a2df31eec93",
      "name": "Chelsea",
    },
    Object {
      "id": "5ff6312e-59d5-462f-bf5e-094834b55e96",
      "name": "Everton",
    },
    Object {
      "id": "bab0f5a9-d202-49f0-ad49-eb49c0d9b701",
      "name": "FC Barcelona",
    },
    Object {
      "id": "6d07e93c-2c15-4de8-85b0-a3a2be741897",
      "name": "Liverpool",
    },
    Object {
      "id": "19d7f679-43d4-4060-b19d-03b30127cf23",
      "name": "Manchester City",
    },
    Object {
      "id": "97243d37-3330-46d3-96df-9513e35a8cc9",
      "name": "Manchester United",
    },
    Object {
      "id": "1412d281-06e0-488d-8795-b46ee26daa4e",
      "name": "Newcastle United",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with filtering should list second page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "id": "cefc7f1b-65ae-44d1-b0a8-79c20b551f41",
      "name": "Tottenham Hotspur",
    },
    Object {
      "id": "e5f9f49c-bf0e-4505-8b3a-030ae3e5d968",
      "name": "Watford",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with filtering should return a bad request if filtered field does not exist 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_BAD_FIELD_ERROR: Unknown column 'team.club' in 'field list'",
  "statusCode": 400,
}
`;

exports[`Teams [GET] /teams with filtering should return no content if out of bounds 1`] = `Object {}`;

exports[`Teams [GET] /teams with limiting should list first page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "07c3eac0-909d-4a1c-9fb5-6a6202c65e34",
      "name": "Arsenal",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "1412d281-06e0-488d-8795-b46ee26daa4e",
      "name": "Newcastle United",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "19d7f679-43d4-4060-b19d-03b30127cf23",
      "name": "Manchester City",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "47362a27-5245-4492-ab5b-8a2df31eec93",
      "name": "Chelsea",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "5ff6312e-59d5-462f-bf5e-094834b55e96",
      "name": "Everton",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
  ],
  "limit": 5,
  "page": 1,
  "pages": 3,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with limiting should list second page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "6d07e93c-2c15-4de8-85b0-a3a2be741897",
      "name": "Liverpool",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "97243d37-3330-46d3-96df-9513e35a8cc9",
      "name": "Manchester United",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "bab0f5a9-d202-49f0-ad49-eb49c0d9b701",
      "name": "FC Barcelona",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "cefc7f1b-65ae-44d1-b0a8-79c20b551f41",
      "name": "Tottenham Hotspur",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "e5f9f49c-bf0e-4505-8b3a-030ae3e5d968",
      "name": "Watford",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
  ],
  "limit": 5,
  "page": 2,
  "pages": 3,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with limiting should return no content if out of bounds 1`] = `Object {}`;

exports[`Teams [GET] /teams with no filtering or sorting should list first page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "07c3eac0-909d-4a1c-9fb5-6a6202c65e34",
      "name": "Arsenal",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "1412d281-06e0-488d-8795-b46ee26daa4e",
      "name": "Newcastle United",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "19d7f679-43d4-4060-b19d-03b30127cf23",
      "name": "Manchester City",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "47362a27-5245-4492-ab5b-8a2df31eec93",
      "name": "Chelsea",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "5ff6312e-59d5-462f-bf5e-094834b55e96",
      "name": "Everton",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "6d07e93c-2c15-4de8-85b0-a3a2be741897",
      "name": "Liverpool",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "97243d37-3330-46d3-96df-9513e35a8cc9",
      "name": "Manchester United",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "bab0f5a9-d202-49f0-ad49-eb49c0d9b701",
      "name": "FC Barcelona",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "cefc7f1b-65ae-44d1-b0a8-79c20b551f41",
      "name": "Tottenham Hotspur",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "e5f9f49c-bf0e-4505-8b3a-030ae3e5d968",
      "name": "Watford",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with no filtering or sorting should list second page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "ebdcacdb-cdd8-4817-b87b-412c3e83ecc0",
      "name": "Bournemouth",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d",
      "name": "AFC Wimbledon",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with no filtering or sorting should return no content if out of bounds 1`] = `Object {}`;

exports[`Teams [GET] /teams with sorting ascending should list first page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d",
      "name": "AFC Wimbledon",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "07c3eac0-909d-4a1c-9fb5-6a6202c65e34",
      "name": "Arsenal",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "ebdcacdb-cdd8-4817-b87b-412c3e83ecc0",
      "name": "Bournemouth",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "47362a27-5245-4492-ab5b-8a2df31eec93",
      "name": "Chelsea",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "5ff6312e-59d5-462f-bf5e-094834b55e96",
      "name": "Everton",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "bab0f5a9-d202-49f0-ad49-eb49c0d9b701",
      "name": "FC Barcelona",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "6d07e93c-2c15-4de8-85b0-a3a2be741897",
      "name": "Liverpool",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "19d7f679-43d4-4060-b19d-03b30127cf23",
      "name": "Manchester City",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "97243d37-3330-46d3-96df-9513e35a8cc9",
      "name": "Manchester United",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "1412d281-06e0-488d-8795-b46ee26daa4e",
      "name": "Newcastle United",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with sorting ascending should list second page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "cefc7f1b-65ae-44d1-b0a8-79c20b551f41",
      "name": "Tottenham Hotspur",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "e5f9f49c-bf0e-4505-8b3a-030ae3e5d968",
      "name": "Watford",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with sorting ascending should return no content if out of bounds 1`] = `Object {}`;

exports[`Teams [GET] /teams with sorting descending should list first page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "e5f9f49c-bf0e-4505-8b3a-030ae3e5d968",
      "name": "Watford",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "cefc7f1b-65ae-44d1-b0a8-79c20b551f41",
      "name": "Tottenham Hotspur",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "1412d281-06e0-488d-8795-b46ee26daa4e",
      "name": "Newcastle United",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "97243d37-3330-46d3-96df-9513e35a8cc9",
      "name": "Manchester United",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "19d7f679-43d4-4060-b19d-03b30127cf23",
      "name": "Manchester City",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "6d07e93c-2c15-4de8-85b0-a3a2be741897",
      "name": "Liverpool",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "bab0f5a9-d202-49f0-ad49-eb49c0d9b701",
      "name": "FC Barcelona",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "5ff6312e-59d5-462f-bf5e-094834b55e96",
      "name": "Everton",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "47362a27-5245-4492-ab5b-8a2df31eec93",
      "name": "Chelsea",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "ebdcacdb-cdd8-4817-b87b-412c3e83ecc0",
      "name": "Bournemouth",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 1,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with sorting descending should list second page of teams with all values 1`] = `
Object {
  "items": Array [
    Object {
      "createdAt": "2018-11-19T00:00:00.000Z",
      "id": "07c3eac0-909d-4a1c-9fb5-6a6202c65e34",
      "name": "Arsenal",
      "updatedAt": "2018-11-19T00:00:00.000Z",
    },
    Object {
      "createdAt": "2018-11-18T00:00:00.000Z",
      "id": "f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d",
      "name": "AFC Wimbledon",
      "updatedAt": "2018-11-18T00:00:00.000Z",
    },
  ],
  "limit": 10,
  "page": 2,
  "pages": 2,
  "total": 12,
}
`;

exports[`Teams [GET] /teams with sorting descending should return no content if out of bounds 1`] = `Object {}`;

exports[`Teams [GET] /teams/:id should return a bad request if filtered field does not exist 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_BAD_FIELD_ERROR: Unknown column 'team.club' in 'field list'",
  "statusCode": 400,
}
`;

exports[`Teams [GET] /teams/:id should return a filtered result 1`] = `
Object {
  "name": "AFC Wimbledon",
}
`;

exports[`Teams [GET] /teams/:id should return a team 1`] = `
Object {
  "createdAt": "2018-11-18T00:00:00.000Z",
  "id": "f3ec4e4b-02bc-4c28-b9ca-ccc6b9f3553d",
  "name": "AFC Wimbledon",
  "updatedAt": "2018-11-18T00:00:00.000Z",
}
`;

exports[`Teams [GET] /teams/:id should return not found if team does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Teams [PATCH] /teams/:id should return not found if team does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;

exports[`Teams [PATCH] /teams/:id should throw an error if name is not a string 1`] = `
Array [
  Object {
    "children": Array [],
    "constraints": Object {
      "isString": "name must be a string",
    },
    "property": "name",
    "target": Object {
      "name": 99,
    },
    "value": 99,
  },
]
`;

exports[`Teams [PATCH] /teams/:id should throw an error of a team is updated with a duplicate name 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_DUP_ENTRY: Duplicate entry 'AFC Wimbledon' for key 'IDX_cf461f5b40cf1a2b8876011e1e'",
  "statusCode": 400,
}
`;

exports[`Teams [PATCH] /teams/:id should update a team 1`] = `Object {}`;

exports[`Teams [POST] /teams should throw an error if a duplicate team is created 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_DUP_ENTRY: Duplicate entry 'Team' for key 'IDX_cf461f5b40cf1a2b8876011e1e'",
  "statusCode": 400,
}
`;

exports[`Teams [POST] /teams should throw an error if name is not a string 1`] = `
Array [
  Object {
    "children": Array [],
    "constraints": Object {
      "isString": "name must be a string",
    },
    "property": "name",
    "target": Object {
      "name": 99,
    },
    "value": 99,
  },
]
`;
